security/pkg/nodeagent/caclient/providers/citadel/client.go

```diff
@@ -93,8 +93,20 @@ func (c *CitadelClient) CSRSign(csrPEM []byte, certValidTTLInSec int64) ([]strin
 		ValidityDuration: certValidTTLInSec,
 		Metadata:         crMetaStruct,
 	}
+	// TODO(hzxuzhonghu): notify caclient rebuilding only when root cert is updated.
+	// It can happen when the istiod dns certs is resigned after root cert is updated,
+	// in this case, the ca grpc client can not automiatically connect to istiod after the underlying network connection closed.
+	// Becase that the grpc client still use the old tls configuration to reconnect to istiod.
+	// So here we need to rebuild the caClient in order to use the new root cert.
+	defer func() {
```

Comment:
It is not possible, the grpc credential options is passed down, so there is not interface to operate on it.

>Its very odd that the function call will leave us with a new connection, but never send a request to it.

Not that, if a error returned, It will return to the envoy too. And envoy will resend the sds request.

Reply:
@howardjohn Is that ok, only rebuild grpc client when this error happen